lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -144,18 +181,21 @@ $SIG{__DIE__} = sub {
 
 sub new {
     my ( $class, %args ) = @_;
-    $args{colour} //= -t STDERR;
+    $args{colour} //= _stderr_is_tty();
     my $self = $class->SUPER::new(sub { }, %args);
-
-    # There are other ways of running containers, but for now "in docker? generate JSON"
-    # is at least a starting point.
-    $self->{in_container} = -r '/.dockerenv';
-    my $json_log_file = $self->{json_log_file};
-    $json_log_file = $0 . '.json.log' if(!$json_log_file && !$self->{in_container});
-    if($json_log_file) {
-        $self->{fh} = path($json_log_file)->opena_utf8 or die 'unable to open log file - ' . $!;
-        $self->{fh}->autoflush(1);
+    # if there is json_log_file, then print json to that file
+    if($self->{json_log_file}) {
+        $self->{json_fh} = path($self->{json_log_file})->opena_utf8 or die 'unable to open log file - ' . $!;
```

Comment:
Before open file could we check for existence,  if the file is not there then touch and give permission here?

Reply:
We needn't. It is opened with `append` mode. So if there is one, then open it , if not, it will be created.